offa.csl

```diff
@@ -0,0 +1,438 @@
+          <text macro="edition"/>
+        </group>
+      </if>
+    </choose>
+  </macro>
+  <macro name="journal-date">
+    <choose>
+      <if type="article-journal">
+        <text macro="date" prefix=", "/>
+      </if>
+    </choose>
+  </macro>
+  <macro name="day-month">
+    <date variable="issued">
+      <date-part name="month"/>
+      <date-part name="day" prefix=" "/>
+    </date>
+  </macro>
+  <macro name="collection-title">
+    <text variable="container-title" prefix=" "/>
+    <text variable="volume" prefix=" "/>
+  </macro>
+  <macro name="event">
+    <text term="presented at" suffix=" "/>
+    <date form="text" variable="issued" suffix=", "/>
+    <text variable="event-place" suffix=". "/>
+    <text variable="event"/>
+  </macro>
+  <macro name="issue">
+    <choose>
+      <if type="speech" match="any">
+        <text macro="event" prefix=" "/>
+      </if>
+      <else-if type="paper-conference" match="any">
+        <text variable="event" prefix=" "/>
+        <date form="text" variable="event-date" prefix=" "/>
+      </else-if>
+      <else-if type="article-newspaper article-magazine" match="any">
+        <text macro="day-month" prefix=", "/>
+      </else-if>
+    </choose>
+  </macro>
+  <macro name="series">
+    <choose>
+      <if match="any" variable="collection-title">
+        <text variable="collection-title" prefix=". "/>
+        <text variable="collection-number" prefix=" "/>
+      </if>
+    </choose>
+    <choose>
+      <if type="paper-conference" match="any">
+        <text variable="volume" prefix=" "/>
+      </if>
+    </choose>
+  </macro>
+  <macro name="url">
+    <choose>
+      <if match="any" variable="URL">
+        <group>
+          <text variable="URL" text-case="lowercase" prefix=". "/>
+          <choose>
+            <if match="none" is-uncertain-date="accessed">
+              <group>
+                <text term="accessed" prefix=" ("/>
+                <date form="numeric" variable="accessed" suffix=")"/>
+              </group>
+            </if>
+          </choose>
+        </group>
+      </if>
+    </choose>
+  </macro>
+  <macro name="doi">
+    <choose>
+      <if match="any" variable="DOI">
+        <group>
+          <text value=". DOI: https://doi.org/"/>
+          <text variable="DOI"/>
+        </group>
+      </if>
+    </choose>
+  </macro>
+  <macro name="review">
+    <choose>
+      <if match="any" variable="reviewed-author">
+        <text term="reviewed-author" form="verb" prefix=", " suffix=" "/>
+        <names variable="reviewed-author">
+          <name delimiter="/" delimiter-precedes-et-al="never" delimiter-precedes-last="always" initialize-with="."/>
+        </names>
+        <text variable="reviewed-title" prefix=", "/>
+      </if>
+    </choose>
+  </macro>
+  <macro name="encyclopedia-article">
+    <text variable="container-title"/>
+    <number prefix=" " variable="volume"/>
+    <choose>
+      <if match="none" is-uncertain-date="issued">
```

Comment:
This is valid, but what's the purposes of this construct? Why would you not print uncertain dates?

Reply:
Again, I am also not sure what this was intended for ...